apps/mobile/src/screens/Onboarding/OnboardingNftSelectorContractScreen.tsx

```diff
@@ -0,0 +1,85 @@
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
+import { useCallback, useMemo } from 'react';
+import { graphql, useLazyLoadQuery } from 'react-relay';
+
+import { NftSelectorContractHeader } from '~/components/NftSelector/NftSelectorContract/NftSelectorContractHeader';
+import { NftSelectorContractPickerGrid } from '~/components/NftSelector/NftSelectorContract/NftSelectorContractPickerGrid';
+import { NftSelectorContractWrapper } from '~/components/NftSelector/NftSelectorContract/NftSelectorContractWrapper';
+import { OnboardingNftSelectorContractScreenQuery } from '~/generated/OnboardingNftSelectorContractScreenQuery.graphql';
+import { MainTabStackNavigatorParamList, MainTabStackNavigatorProp } from '~/navigation/types';
+
+import { useProfilePicture } from '../NftSelectorScreen/useProfilePicture';
+
+export function OnboardingNftSelectorContractScreen() {
```

Comment:
I love separating these out into different components depending on usage, but it looks like `OnboardingNftSelectorContractScreen` and `PfpSelectorContractScreen` are 99% the same code.Is it possible to consolidate the shared code a bit more? 

I'm imagining `OnboardingNftSelectorContractScreen` and `PfpSelectorContractScreen` to be very thin wrappers around a shared component that contains most of the code that's the same, and the wrappers only add the unique things like the `fullscreen` prop or the custom `handleSelectNft` handler in the case of `PostNftSelectorContractScreen`. 

the main concern would be having to maintain the same logic in 3 different places, and repeating it again when we add another variant of the SelectorContractScreen component

Reply:
Same comment for `SelectorScreen` components like `PostNftSelectorScreen`, `PfpSelectorScreen`,

Reply:
That make sense! Clean up majority of the code so that no redundant code